### DISCUSSION TITLE
freeze an ally in a spot

### DISCUSSION THREAD

**Discussion by androabhi66** [flareteam/flare-game#980] [Q&A] [answered]
hi to all, i'm new, 
is there a way to create a turret (ally) that doesn't move?
even if i create an allied npc (turret-shaped) with speed=0 it follows me if i move too far away
is there a way to "freeze" an ally in one spot? (also with a "freezing" power that keeps it still in that spot... maybe)
thanks and sorry if the question may seem stupid

**Comment by dorkster** [comment, marked as answer]
I would go as far as to consider this a bug. I think it'd be safe to assume that if an entity's speed is 0, it shouldn't be able to follow the player in that way. There's actually 3 scenarios where an immobile summoned entity can "teleport":

1. When too far from the summoner
2. When the summoner moves through an `intramap` event (teleport within the same map)
3. When the summoner moves through an `intermap` event (teleport to another map)

I think it can be safe to say that for 1 & 2, the summon shouldn't ever move with zero speed. Not sure about 3, though. I think we'd want to first determine if the summon's speed is naturally zero or if it is due to a temporary debuff. For the former, we should probably kill off the summon on map transition. Otherwise, we can have it follow the player as other summons would.

**Reply by androabhi66** [reply]
Point 1 describes my current situation.
And I have no idea how to solve it, not even by inventing some kind of "block".
At some point all the turrets and waypoints follow me if I move too far away, making it impossible for me to play the way I want to play.
[light_turret1.txt](https://github.com/user-attachments/files/18009640/light_turret1.txt)
I'm posting the example file to show that the speed is 0, and to see if I'm the one doing something wrong.

**Reply by dorkster** [reply]
You're not doing anything wrong. It's just something that needs to be fixed on the engine side.

**Reply by androabhi66** [reply]
ok, thank you very much.
then i think I just have to wait.
